### PR TITLE
Npgsql 6 upgrade

### PR DESCRIPTION
**Pull request by oskardudycz**
- Added NpgsqlTypeMapping as a clone of the current Npgsql 6 mappings based on https://github.com/npgsql/npgsql/blob/a1d366c5692cc00a5edc1ec5bc9090952c7a63e7/src/Npgsql/TypeMapping/BuiltInTypeHandlerResolver.cs

See related PR in Marten: https://github.com/JasperFx/marten/pull/1987

### CODE REVIEW

File: src/Weasel.Postgresql/Weasel.Postgresql.csproj
Comment (luber) [inline src/Weasel.Postgresql/Weasel.Postgresql.csproj:21]: they have released v6.0.0 already

Comment (oskardudycz) [reply src/Weasel.Postgresql/Weasel.Postgresql.csproj:21]: Thank you for giving the heads up. We're aware of the release. However, it has many breaking changes in the core parts that Marten/Weasel depends on. We're discussing with the Npgsql team if we could find some solution. If not, then we'll have to workaround that.

**Review by mysticmind — APPROVED** [review]
Looks good to me @oskardudycz 👍

**Comment by mizrael** [comment]
hi, I'm waiting for this PR to be merged, I would like to use Marten in a .NET6 project but it's currently not possible due to the dependency on Npgsql 5.0.10.

**Comment by oskardudycz** [comment]
@mizrael we want to add also a few other changes before this PR (it’ll take about a week or two). If you’d like to, I can create prerelease version.

**Comment by mizrael** [comment]
thanks @oskardudycz . Unfortunately I am referencing Npgsql already in other parts of the project. I can totally wait for the final version (it's a pet project, no rush for now), in the meantime I could try to refactor and get rid of the dependency.
thanks!

**Comment by kieranbenton** [comment]
If producing a pre release version is not too much trouble it would be very useful for us and would allow us to test in parallel.

**Comment by samisq** [comment]
I'd like to echo the same sentiment - if it's possible to publish a pre-release version of Marten with Npgsql v6 support, that would be much appreciate, as it would unblock our .net 6/npgsql 6 upgrade testing

**Comment by oskardudycz** [comment]
@samisq, @mizrael, @kieranbenton, I apologise for the long time not giving you alpha, but here it is: https://www.nuget.org/packages/Marten/5.0.0-alpha.3. Real v5 should be in the next few weeks.

**Pull request merged**
